### PR TITLE
ARROW-13249: [Java][CI] Consistent timeout in the Java JNI build

### PR DESCRIPTION
**Pull request by kszucs**
The JNI build gets stopped due to build timeout. Seems like the docker cache isn't valid anymore so it must build the docker image as well, but doesn't have the opportunity to push at the and of the build.

### CODE REVIEW

**Comment by pitrou** [comment]
As you can see, after bumping the timeout a lot, it fails with another error. I would suggest sending an e-mail to the dev mailing-list about JNI CI, and if nobody answers, disable the build.

File: .github/workflows/java_jni.yml [hunk 1]
```diff
@@ -50,15 +50,7 @@ jobs:
     name: AMD64 Debian 9 Java JNI (Gandiva, Plasma, ORC, Dataset)
     runs-on: ubuntu-latest
     if: ${{ !contains(github.event.pull_request.title, 'WIP') }}
-    timeout-minutes: 40
-    strategy:
-      fail-fast: false
-      matrix:
-        jdk: [8]
-        maven: [3.5.2]
```
Comment (kszucs) [inline .github/workflows/java_jni.yml:58]: The default in `.env` is `3.5.4`

**Comment by kszucs** [comment]
After using maven 3.5.4 the build work again. I'm going to reduce the build timeout to 90 minutes so it can rebuild the image on a cache miss.

**Review by kszucs — COMMENTED** [review]
+1

**Comment by pachadotdev** [comment]
@github-actions crossbow submit test-r-linux-valgrind

**Pull request closed without merging**